spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -277,17 +277,15 @@ public void commit(WriterCommitMessage[] messages) {
       IsolationLevel isolationLevel = writeConf.isolationLevel();
       Long validateFromSnapshotId = writeConf.validateFromSnapshotId();
 
+      if (isolationLevel != null && validateFromSnapshotId != null) {
+        dynamicOverwrite.validateFromSnapshot(validateFromSnapshotId);
+      }
+
       if (isolationLevel == SERIALIZABLE) {
-        if (validateFromSnapshotId != null) {
-          dynamicOverwrite.validateFromSnapshot(validateFromSnapshotId);
-        }
         dynamicOverwrite.validateNoConflictingData();
```

Comment:
Nit: It seems like this one is repeated in both cases. Since there's no existing `else` clause, it's probably not worth reducing the duplication as we might need more `if` statements but thought I'd call it out.

Reply:
Thanks yea good catch, I was thinking to leave it here to make it more clear in each validation mode what we are validating, and maybe there are future modes where we do not do this as you said.  As opposed to the validationSnapshotId which is the basis of all validations.